pkg/assembler/clients/helpers/bulk.go

```diff
@@ -615,3 +634,48 @@ func ingestIsOccurrences(ctx context.Context, client graphql.Client, v []assembl
 	}
 	return nil
 }
+
+func ingestCertifyLegals(ctx context.Context, client graphql.Client, v []assembler.CertifyLegalIngest) error {
+	var pkgs []model.PkgInputSpec
+	var sources []model.SourceInputSpec
+	var pkgDec [][]model.LicenseInputSpec
+	var pkgDis [][]model.LicenseInputSpec
+	var pkgCL []model.CertifyLegalInputSpec
+	var srcDec [][]model.LicenseInputSpec
+	var srcDis [][]model.LicenseInputSpec
+	var srcCL []model.CertifyLegalInputSpec
+	for _, ingest := range v {
+
+		if ingest.Pkg != nil && ingest.Src != nil {
```

Comment:
we can add these checks to the resolver but you already have that as part of your issue TODOs so all good.